### DISCUSSION TITLE
[Feature Request] remove an entity.

### DISCUSSION THREAD

**Discussion by janusn** [frenck/spook#864] [Documentation]
Is it possible to delete an entity by automation? There are outdated entities generated from an integration periodically. I would like to remove them on a periodic schedule by automation.

> Background: I have [Glances integration](https://www.home-assistant.io/integrations/glances/) installed. It creates a couple sensors for each network interface of the target machine. There are docker containers running on the target machines. As I frequently pull and recreate containers for updated images, new network sensors are created in HASS for the new bridge networks. The problem is, the integration does not remove deleted networks on HASS as they are removed. Hence there are hundreds of unavailable sensors for the devices under Glances integration. I have filed a [feature request](https://github.com/home-assistant/core/issues/124563) on the project. But it seems the developer does not consider implementing it.

**Comment by frenck** [comment]
The problem here is, adjusting automations manually can be tricky. If a user added custom YAML stuff, like comments, formatting, anchors and what not; migrating those automations by code will lose all the custom stuff.

For that reason, this is a no go for me.

../Frenck

**Reply by janusn** [reply]
Thanks for your reply.

I understand your concern. However as far as I know, Home Assistant does not modify configuration.yaml. Hence all the mistakes made to the entities defined in YAML can be recovered easily by restarting the Home Assistant OS/container. And thanks to the brilliant feature, Repairs of Spook, a mistake can be spotted easily.

**Reply by PavelD** [reply]
HI @frenck
I'm searching for the same. I have many entities created by appdaemon and some become obsolete. The second source of entities is script `python_script.set_state`. There is many ways hoe to create entity not linked to any integration and it will be really handy to have some way how to delete entities without unique id.

I see your concern is consistency. What about to create 2-3 different services?

1st service to delete any entity
2st service to find all places where is entity used or customized. Any non-empty result of this service can fail execution of the 1st service.
3rd service to find all places where is customization / usage of non existing service.

I know non existing entity used in UI is reported already. And deletion is already in place, but just for orphan entities.

**Reply by janusn** [reply]
There are tons of these obsolete entities in my setup.
![Screenshot 2024-11-20 at 15 33 23](https://github.com/user-attachments/assets/a2585d44-0e11-4b0f-b268-a713a312bad6)

I have tried to bulk delete them from the Settings->Entities. Unfortunately, they don't show up from the unavailable filter.
![Screenshot 2024-11-20 at 15 36 34](https://github.com/user-attachments/assets/61ef7ebb-b01c-49f3-905c-6dd56ffdcb75)

 Nor can they be deleted. Applying delete from that page does nothing.

**Comment by stevegroom** [comment]
Another problematic entity I have seen is when a Matter device goes away and you retrigger the pairing without removing the old device first. The device starts to work again, but with two sets of entities - the old ones are unavailable and cannot be deleted.
